[BUG] civo kubernetes node pool does not scale or update the node pool from terraform provider

### Description

## 1. What's the issue?

## 1.1 Description

When trying to update the amount of nodes on a `civo_kubernetes_node_pool` resource, the provider throws an exception, and the node pool is never created, also, when the node is added from console it does not add the labels and/or taints as the other node in the same node pool.

## 1.2 Screenshots


## 2. Steps to rexyzuce

```
$[~] tofu plan -out=tofu.plan
module.xyz.civo_reserved_ip.this[0]: Refreshing state... 
module.xyz.civo_network.this[0]: Refreshing state... 
module.xyz.civo_firewall.this[0]: Refreshing state... 
module.xyz.civo_database.this["xyzdb"]: Refreshing state... 
module.xyz.civo_kubernetes_cluster.this[0]: Refreshing state... 
module.xyz.civo_kubernetes_node_pool.pool["xyz"]: Refreshing state... 
                                                                                                                                                                                                                                              
OpenTofu used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:                                                                                                     
  ~ update in-place                                                                                                                                                                                                                           
                                                                                                                                                                                                                                              
OpenTofu will perform the following actions:                                                                                                                                                                                                  
                                                                                                                                                                                                                                              
  # module.xyz.civo_kubernetes_cluster.this[0] will be updated in-place                                                                                                                                                                      
  ~ resource "civo_kubernetes_cluster" "this" {                                                                                                                                                                                               
        id                     = "999999-9999-9999-9999-99999999999999"                                                                                                                                                                       
        name                   = "xyz-cluster"                                                                                                                                                                                               
        # (16 unchanged attributes hidden)                                                                                                                                                                                                    
                                                                                                                                                                                                                                              
      ~ pools {                                                                                                                                                                                                                               
          ~ labels              = {                                                                                                                                                                                                           
              + "environment" = "nonxyz"                                                                                                                                                                                                     
            }                                                                                                                                                                                                                                 
            # (5 unchanged attributes hidden)                                                                                                                                                                                                 
        }                                                                                                                                                                                                                                     
    }                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                              
  # module.xyz.civo_kubernetes_node_pool.pool["xyz"] will be updated in-place                                                                                                                                                               
  ~ resource "civo_kubernetes_node_pool" "pool" {                                                                                                                                                                                             
        id             = "999999-9999-9999-9999-99999999999999"                                                                                                                                                                               
      ~ node_count     = 1 -> 2                                                                                                                                                                                                               
        # (4 unchanged attributes hidden)                                                                                                                                                                                                     
                                                                                                                                                                                                                                              
        # (1 unchanged block hidden)                                                                                                                                                                                                          
    }                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                              
Plan: 0 to add, 2 to change, 0 to destroy.                                                                                                                                                                                                    
                                                                                                                                                                                                                                              
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 
                                                                                                                                                                                                                                              
Saved the plan to: tofu.plan                                                                                                                                                                                                                  
                                                                                                                                                                                                                                              
To perform exactly these actions, run the following command to apply:                                                                                                                                                                         
    tofu apply "tofu.plan"                               

```

Then when you try to apply, you get:

```

tofu apply tofu.plan                                                                                                                                    *[feature/upgrade-flux-2-7] 
module.xyz.civo_kubernetes_cluster.this[0]: Modifying... 
module.xyz.civo_kubernetes_cluster.this[0]: Still modifying... 
module.xyz.civo_kubernetes_cluster.this[0]: Modifications complete after 12s 
module.xyz.civo_kubernetes_node_pool.pool["xyz"]: Modifying... 
╷                                                                                                                                                                                                                                             
│ Error: Plugin did not respond
│ 
│   with module.xyz.civo_kubernetes_node_pool.pool["xyz"],
│   on .terraform/modules/xyz/main.tf line 80, in resource "civo_kubernetes_node_pool" "pool":
│   80: resource "civo_kubernetes_node_pool" "pool" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-civo_v1.1.7 plugin: 

panic: interface conversion: interface {} is int, not *int

goroutine 117 [running]:
github.com/civo/terraform-provider-civo/civo/kubernetes.resourceKubernetesClusterNodePoolUpdate({0x131ddb0, 0xc00056d1f0}, 0xc0000ad700, {0x1144140?, 0xc0000c4c30?})
        /home/runner/work/terraform-provider-civo/terraform-provider-civo/civo/kubernetes/resource_kubernetes_cluster_nodepool.go:214 +0xf14
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000487ce0, {0x131dd08, 0xc00027bb60}, 0xd?, {0x1144140, 0xc0000c4c30})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:812 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000487ce0, {0x131dd08, 0xc00027bb60}, 0xc000192dd0, 0xc0000ad480, {0x1144140, 0xc0000c4c30})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00045ec60, {0x131dd08?, 0xc00027ba70?}, 0xc000592280)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0003532c0, {0x131dd08?, 0xc00027b1d0?}, 0xc00056c0e0)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/tf5server/server.go:859 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1103260?, 0xc0003532c0}, {0x131dd08, 0xc00027b1d0}, 0xc0000ac400, 0x0)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:503 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003a6b40, {0x131dd08, 0xc00027b140}, {0x1322c20, 0xc000102340}, 0xc0005b2c60, 0xc00049c210, 0x1aef0f8, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1372 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0003a6b40, {0x1322c20, 0xc000102340}, 0xc0005b2c60)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1783 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1016 +0x59
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 35
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1027 +0x115

Error: The terraform-provider-civo_v1.1.7 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```



### Acceptance Criteria

## Functional Acceptance Criteria

### 1. Node Pool Scaling
- Updating the `node_count` attribute in a `civo_kubernetes_node_pool` resource must successfully add or remove nodes without causing provider crashes or panics.
- The provider must correctly handle integer conversions internally (no `interface conversion` errors).
- The resource update should complete successfully and reflect the desired node count in both Terraform state and the Civo cluster.

### 2. Label and Taint Propagation
- When a node is added (either via Terraform or the Civo Console), it must inherit the same labels and taints defined in the existing node pool.
- All nodes within a node pool must maintain consistent labels and taints after any scale-up or scale-down operation.
- A subsequent `tofu plan` after successful apply should show no drift in labels or taints.

### 3. Error Handling
- If an update cannot be performed due to an API or validation issue, the provider must return a clear error message rather than panic.
- The provider must gracefully handle partial updates or failed scaling attempts and preserve a consistent Terraform state.

### 4. Backward Compatibility
- Existing node pools created prior to the fix must continue to support updates to `node_count`, `labels`, and `taints` without requiring re-creation.
- No behavior changes should occur for other unrelated Civo Kubernetes resources.

--- a/civo/kubernetes/resource_kubernetes_cluster_nodepool.go
+++ b/civo/kubernetes/resource_kubernetes_cluster_nodepool.go
@@ -211,7 +211,8 @@ func resourceKubernetesClusterNodePoolUpdate(ctx context.Context, d *schema.Reso
 	}
 
 	if d.HasChange("node_count") {
-		poolUpdate.Count = d.Get("node_count").(*int)
+		count := d.Get("node_count").(int)
+		poolUpdate.Count = &count
 	}
 
 	if d.HasChange("labels") {

